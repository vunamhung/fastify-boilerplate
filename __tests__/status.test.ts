import fastify, { FastifyInstance } from 'fastify';
import { Server, IncomingMessage, ServerResponse } from 'http';

import statusRoutes from '../src/routes/status';

describe('/', () => {
  let server: FastifyInstance<Server, IncomingMessage, ServerResponse>;

  beforeAll(() => {});

  beforeEach(async () => {
    server = fastify({});
    server.register(statusRoutes);
    await server.ready();

    jest.clearAllMocks();
  });

  it('GET returns 200', async (done) => {
    const response = await server.inject({ method: 'GET', url: '/' });
    expect(response.statusCode).toEqual(200);
    const payload: { date: Date; works: boolean } = JSON.parse(response.payload);
    expect(payload).toMatchSnapshot({ date: expect.any(String), works: true });

    done();
  });

  it('POST returns 404', async (done) => {
    const response = await server.inject({ method: 'POST', url: '/' });
    expect(response.statusCode).toEqual(404);
    expect(response.payload).toMatchSnapshot();

    done();
  });
});
