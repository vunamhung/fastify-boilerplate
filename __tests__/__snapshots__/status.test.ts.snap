// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ GET returns 200 1`] = `
Object {
  "date": Any<String>,
  "works": true,
}
`;

exports[`/ POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/ not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
